test/controllers/sections/transportUnit/TransportSealTypeControllerSpec.scala

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.sections.transportUnit
+
+import base.SpecBase
+import fixtures.TransportUnitFixtures
+import forms.TransportSealTypeFormProvider
+import mocks.services.MockUserAnswersService
+import models.TransportUnitType.Container
+import models.{NormalMode, TransportUnitType, UserAnswers}
+import navigation.FakeNavigators.FakeNavigator
+import navigation.Navigator
+import pages.TransportUnitTypePage
+import pages.sections.transportUnit.TransportSealTypePage
+import play.api.inject.bind
+import play.api.mvc.Call
+import play.api.test.FakeRequest
+import play.api.test.Helpers._
+import services.UserAnswersService
+import views.html.sections.transportUnit.TransportSealTypeView
+
+import scala.concurrent.Future
+
+class TransportSealTypeControllerSpec extends SpecBase with MockUserAnswersService with TransportUnitFixtures {
+
+  class Fixture(userAnswers: Option[UserAnswers] = Some(emptyUserAnswers)) {
+
+    def onwardRoute = Call("GET", "/foo")
+
+    val formProvider = new TransportSealTypeFormProvider()
+    val form = formProvider()
+
+    lazy val transportSealTypeRoute = routes.TransportSealTypeController.onPageLoad(testErn, testLrn, NormalMode).url
+
+    val application = applicationBuilder(userAnswers)
+      .overrides(
+        bind[Navigator].toInstance(new FakeNavigator(onwardRoute)),
+        bind[UserAnswersService].toInstance(mockUserAnswersService)
+      )
+      .build()
+
+    val view = application.injector.instanceOf[TransportSealTypeView]
+  }
+
+
+  "TransportSealType Controller" - {
+
+    "must return OK and the correct view for a GET" in new Fixture(Some(emptyUserAnswers
+      .set(TransportUnitTypePage, Container)
+    )) {
+
+      running(application) {
+
+        val request = FakeRequest(GET, transportSealTypeRoute)
+
+        val result = route(application, request).value
+
+        status(result) mustEqual OK
+        contentAsString(result) mustEqual view(
+          form = form,
+          transportUnitType = TransportUnitType.Container,
+          onSubmitCall = controllers.sections.transportUnit.routes.TransportSealTypeController.onSubmit(testErn, testLrn, NormalMode)
+        )(dataRequest(request), messages(application)).toString
+      }
+    }
+
+    "must populate the view correctly on a GET when the question has previously been answered" in new Fixture(Some(emptyUserAnswers
```

Comment:
Is it worth having a test to ensure the page is populated when only the seal type has been entered?